File: observer/scala/src/main/scala/Subscriber.scala
package observer

class Subscriber(val publisher: Publisher) {
  var subscription: Int = _

  def shutdown(): Unit = {
    println("Subscriber: Unsubscribing...")
    publisher.unsubscribe(subscription)
  }

  def speak(d: String): Unit = println(s"Subscriber: Publisher says ${'"'}$d${'"'}")
}

File: observer/go/src/subscriber.go
package src

import "fmt"

type Subscriber struct {
	publisher    *Publisher
	Subscription int
}

func (s *Subscriber) Init(p *Publisher) {
	s.publisher = p
}

func (s *Subscriber) Shutdown() {
	s.publisher.Unsubscribe(s.Subscription)
}

func (s *Subscriber) Speak() {
	fmt.Println("Hello")
}
